pandas/core/frame.py

```diff
@@ -8394,6 +8394,52 @@ def isin(self, values):
                 self.columns,
             )
 
+    def value_counts(self):
+        """
+        The number of times each unique row appears in the DataFrame.
+
+        Rows that contain any NaN value are omitted from the results.
+
+        Returns
+        -------
+        counts : Series
+
```

Comment:
it might be appropriate to add a subset argument here to select the columns to use (similar to what we do for dropna) but could always be added later